draft-ietf-taps-arch.md

```diff
@@ -311,22 +311,17 @@ The diagram below provides a high-level view of the actions taken during the lif
      +---------------+ Initiate() +------------+ Abort()   :
  +-->| Preconnection |----------->| Connection |---------------> Closed
  |   +---------------+     :      +------------+ Connection:
- |                         :      ^   ^    |     Finished  :
- +-- Local Endpoint        :      |   |    |               :
- |                         :      |   |    +---------+     :
- +-- Remote Endpoint       :      |   |              |     :
- |                         :      |   |Send()        |     :
- +-- Path Selection        :      | +---------+      v     :
- |   Properties            :      | | Message |  Message   :
- |                         :      | | to send |  Received  :
- +-- Protocol Selection    :      | +---------+            :
- |   Properties            :      |                        :
- |                         :      |                        :
- +-- Specific Protocol     :      |                        :
- |   Properties            :      |                        :
- |                         :      |                        :
- |   +----------+          :      |                        :
- +-->| Listener |-----------------+                        :
+ |                         :       ^  ^  ^  |    Finished  :
+ +-- Local Endpoint        :       |  |  |  |              :
+ |                         :       |  |  |  +---------+    :
+ +-- Remote Endpoint       :  +----+  |  +-----+      |    :
+ |                         :  |       |Send()  |      |    :
+ +-- Selection Properties  :  |  +---------+   |      v    :
+ +-- Connection Properties ---+  | Message |   |  Message  :
+ +-- Message Properties -------->| to send |   |  Received :
```

Comment:
why are Message Properties bound to Preconnections? The availability of some Message Properties may depend on Selection Properties, and one can set default Message Properties, but this rendering makes it look like setting defualt Message Properties on a Preconnection is non-optional...

Reply:
We do not distinguish between optional and mandatory in the drawing at all, e.g., we have listen and imitate paths in there.

The only problem I see is that we might have to clarify in API that "default message properties specified on a pre-connection that do not apply do not cause errors – if you need the property, set the corresponding selection property to require "

Reply:
+1, it makes sense to note this at the level of the archietcture; can you add this sentence when doing the sublist change above, then I'll merge...